iceoryx_utils/include/iceoryx_utils/cxx/make_scoped_static.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
```

Comment:
Could you rename this file into `scoped_static.hpp` and maybe create an alias `using ScopedStatic = GenericRAII`.
The function `make_scoped_static` is then the factory to create a `ScopedStatic`.

Reply:
Renamed the file.

@elBoberido IIUC, `makeScopedStatic` goes hand-in-hand with `GenericRAII`? Generalising would mean our `ScopedStatic` would require taking one callable, which for me is mainly `GenericRAII`.

Reply:
I'm not sure what you mean? If you suggest to remove the dependency to GenericRAII, go for it.

I would also not create the `using ScopedStatic = GenericRAII` alias since it would indicate that the return value is a `ScopedStatic`, which it is not. The scoped static is the `cxx::optional<T>` or any other container which has a `emplace` and `destroy` method. A better name is welcome.

A potential solution would be
```c++
template <typename T>
class ScopedStaticGuard {
  public:
    ~ScopedStaticGuard() { m_meory.reset(); }
  private:
    friend .... makeScopedStatic
    ScopedStaticGuard(T& memory) : m_memory(memory) {}

    T& m_memory;
};

template <typename T, typename... CTorArgs>
static ScopedStaticGuard makeScopedStatic(T& memory, CTorArgs&&... ctorArgs) {
    memory.emplace(std::forward<CTorArgs>(ctorArgs)...);
    return ScopedStaticGuard<T>{memory};
}
```

Reply:
Ok, we could consider this. But that's something for another pull request.